pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -312,50 +392,18 @@ func (e *DockercfgController) createTokenSecret(serviceAccount *api.ServiceAccou
 			Annotations: map[string]string{
 				api.ServiceAccountNameKey: serviceAccount.Name,
 				api.ServiceAccountUIDKey:  string(serviceAccount.UID),
+				PendingDockercfgNameKey:   "true",
```

Comment:
@deads2k @liggitt I rewrote that add/update logic so the annotation is no longer needed and is is much cleaner now.